applications/nrf_desktop/src/modules/dvfs.c

```diff
@@ -309,8 +321,10 @@ static bool app_event_handler(const struct app_event_header *aeh)
 		process_dvfs_states(DVFS_STATE_CONFIG_CHANNEL, true);
 		struct dvfs_state_timeout *config_channel_timeout =
 			&dvfs_state_timeouts[DVFS_STATE_CONFIG_CHANNEL];
-		(void) k_work_reschedule(&(config_channel_timeout->timeout_work),
-					 K_MSEC(config_channel_timeout->timeout_ms));
+		if (config_channel_timeout->timeout_ms > 0) {
```

Comment:
Maybe we could move it into `process_dvfs_states` (to make it part of generic implementation)

Reply:
we already discussed it and came to conclusion that k_work_reschedule stays here and so this check should stay here.

Reply:
I recall that. I suggested update here to make the implementation more generic (the change could fit nicely together with newly introduced Kconfigs - leaving it as is looks a bit strange to me). Current version also increases risk of misalignment between Kconfig configuration and C code (e.g. if someone adds or removes timeout somewhere) for actually no reason. Still, I would not insist.